src/docs/show-help-files/help-prte-runtime.txt

```diff
@@ -144,3 +144,43 @@ other nodes using the "prte_remote_tmpdir_base" param.
 This is only a warning advisory and your job will continue. You can
 disable this warning in the future by setting the
 "prte_silence_shared_fs" MCA param to "1".
+
+[prte:slurm:envar]
+PRTE detected the presence of an MCA parameter in the environment that
+assigns custom command line arguments to the `srun` command used to
+start PRTE's daemons on remote nodes:
+
+  Paramater value: %s
+
+This warning is provided to alert you (the user) that this parameter
+value will be ignored.
+
+Background: Starting with Slurm version 23.11, a command line argument
+(`--external-launcher`) was added to `srun` to indicate that the
+command was being initiated from within a third-party launcher (e.g.,
+`prte` or `prterun`). This allows Slurm to essentially freely modify
+the `srun` command line while retaining a backward compatibility
+capability when explicitly told to use it.  Notably, Slurm
+did this by automatically setting the PRTE_MCA_plm_slurm_args environment
+variable to pass in its own command line arguments.
+
+Unfortunately, this had the side effect of overriding most user- or
+system-level settings. In addition, arguments passed on the
+PRTE command line overrode any Slurm setting of the environment
+variable, but with potentially undesirable side effects if newer
+versions of `srun` misinterpret or fail to understand the user-specified
+arguments.
+
+PRRTE now directly determines the `srun` version and sets its `srun` cmd
+line arguments accordingly. These arguments are set in addition to any
+provided by the user, and therefore the user no longer needs to concern
+themselves with changes induced by Slurm.
+
+If the setting of the MCA parameter was intentional, or if the
+parameter value looks acceptable to you, then please set the MCA parameter
+either in the default MCA param file (either the system or user-level file),
+or on the cmd line:
+
+  Cmd line: --prtemca plm_slurm_args %s
+  Default MCA param file: plm_slurm_args = %s
+
```

Comment:
Couple of issues here. First, the value we can see is the _final_ envar value - i.e., if Slurm is going to overwrite it, they already have done so.  We have no idea if the user attempted to set an envar for that param, nor what that value might have been.

Second, we still need to address the problem this PR creates - it will _block_ any execution under Slurm 23.11 and beyond. I do think that (if we are going to do this) we need some flag the user can set to indicate that we should ignore the envar value (i.e., unsetenv it) and continue executing.

Reply:
> it will block any execution under Slurm 23.11 and beyond.

Oh, gotcha -- no, I don't think we want that.  I thought we talked about adding (abstraction-breaking, but probably still necessary) code in the tools codes directly -- before `prte_init()` and friends...?

That's why I was thinking we could show the value and have good confidence that it's the user's value -- not the SLURM-set value.

Am I thinking incorrectly?

Reply:
> Are you saying that this will happen:
> 
> ```shell
> shell$ salloc -N 1
> shell$ echo $PRTE_MCA_plm_slurm_args
> ...will show the SLURM-set value here...
> ```

Pretty sure this is true

> Or does SLURM set it in every allocation and also set it after mpirun and before prte (i.e., in the remote 

Pretty sure this is not true, but I have no way of testing it.

>  I.e., in all possible cases, SLURM sets this env variable, and it's just 100% not possible for the user to overwrite it.

I don't believe this is true, but I honestly am not sure. There fix was aimed at `mpirun` users, but IIUC the envar was set at time-of-allocation. I don't see any way to set it between the `mpirun` and fork/exec of `prterun` as Slurm isn't involved there. So it would seem like users _could_ override it if they chose, so long as they realized they need to include the new cmd line option.

The issue, in my mind, is that the envar is overriding what users put in their system and user-level default param files because the envar gets processed _after_ those files. It isn't a question of someone setting an envar that doesn't work - it's a problem of putting values into default param files and unexplainingly finding them ignored. The user didn't put anything in the environment - their set values just "disappear" somehow.

Not sure what to do with it. Still a puzzlement on right path.

Reply:
Yeah, the problem is that it is impossible to tell who set the envar.

Only thing I can think of to try is to check for the envar and then parse the value if it is found. We know what Slurm is setting - so if we see that value, we could silently ignore it. If we see any other value, then we probably can safely assume that the user provided it, and let the MCA system handle it as usual.

Would take a little checking to ensure we correctly identify the Slurm value. I can give that a try.

Reply:
Afraid I cannot find a new enough version of Slurm I can put in my container - and as reported elsewhere, I'm having zero success getting Slurm to run in the containers. I give up. Let's just silently ignore it?